src/frontend/src/scheduler/plan_fragmenter.rs

```diff
@@ -467,19 +651,40 @@ impl StageGraphBuilder {
 
 impl BatchPlanFragmenter {
     /// Split the plan node into each stages, based on exchange node.
-    pub fn split(mut self, batch_node: PlanRef) -> SchedulerResult<Query> {
-        let root_stage = self.new_stage(batch_node, Distribution::Single.to_prost(1, &self))?;
-        let stage_graph = self.stage_graph_builder.build(root_stage.id);
+    pub fn split(&mut self, batch_node: PlanRef) -> SchedulerResult<()> {
```

Comment:
The api is not safe here. In fact, we can merge the `split` method with `new` method here since it's meaningless to keep a standalone `split` method.